adding Sixel Support to PwshSpectreConsole

trackd opened the pull request:
Sixel

![image](https://github.com/user-attachments/assets/668863c9-dbe2-42a9-b97e-d6db9963e46b)

still needs a bit of work but most things seem to render ok..

need to investigate transparency a bit more.

ShaunLawrie commented on PwshSpectreConsole/private/classes/PwshSpectreConsole.Sixel.cs:
It'd be worth dragging out the magic characters into some const values with names like SIXEL_CR / SIXEL_LF or something so it makes more sense what these are.

trackd commented:
```md
| Method        | Mean     | Error    | StdDev   | Median   | Gen0      | Gen1      | Gen2      | Allocated |
|-------------- |---------:|---------:|---------:|---------:|----------:|----------:|----------:|----------:|
| ImgToSixelOld | 34.86 ms | 0.693 ms | 1.250 ms | 34.25 ms | 1500.0000 | 1031.2500 | 1000.0000 |   7.57 MB |
| ImgToSixelNew | 27.59 ms | 0.516 ms | 0.483 ms | 27.44 ms |  968.7500 |  968.7500 |  968.7500 |   1.19 MB |

```

converted it to stringbuilder.. seems alot better :)
also refactored the class a bit, it's not broken up into much smaller methods which will simplify when adding gif/animated sixels.

ShaunLawrie commented:
Most of this code is upstream in https://github.com/spectreconsole/spectre.console/compare/main...ShaunLawrie:spectre.console:main

It's in prerelease atm (just copied the dlls into the repo) and it's full of hacks and visual bugs lol https://prerelease.pwshspectreconsole.com/reference/images/get-spectresixelimage/

![image](https://github.com/user-attachments/assets/3b7be306-0bc3-4399-ba3f-823d7edde418)

trackd commented:
closing this as we broke it out into it's own [sixel module](https://github.com/trackd/sixel) and adding the code upstream to SpectreConsole.

will push the non-sixel stuff in a separate pr
